Source/WebGPU/WebGPU/RenderBundleEncoder.mm

```diff
@@ -150,14 +167,19 @@ static void addResource(RenderBundle::ResourcesContainer* resources, id<MTLResou
     id<MTLIndirectRenderCommand> icbCommand = currentRenderCommand();
     if (!icbCommand)
         return;
-
+#if ENABLE(WEBGPU_USE_ICB_REPLAY)
     [icbCommand setRenderPipelineState:m_currentPipelineState];
+#endif
 
-    for (size_t i = 0, sz = m_vertexBuffers.size(); i < sz; ++i)
-        [icbCommand setVertexBuffer:m_vertexBuffers[i].buffer offset:m_vertexBuffers[i].offset atIndex:i];
+    for (size_t i = 0, sz = m_vertexBuffers.size(); i < sz; ++i) {
+        if (m_vertexBuffers[i].buffer)
+            [icbCommand setVertexBuffer:m_vertexBuffers[i].buffer offset:m_vertexBuffers[i].offset atIndex:i];
+    }
 
-    for (size_t i = 0, sz = m_fragmentBuffers.size(); i < sz; ++i)
-        [icbCommand setFragmentBuffer:m_fragmentBuffers[i].buffer offset:m_fragmentBuffers[i].offset atIndex:i];
+    for (size_t i = 0, sz = m_fragmentBuffers.size(); i < sz; ++i) {
+        if (m_fragmentBuffers[i].buffer)
+            [icbCommand setFragmentBuffer:m_fragmentBuffers[i].buffer offset:m_fragmentBuffers[i].offset atIndex:i];
+    }
```

Comment:
Should this also be guarded by `#if ENABLE(WEBGPU_USE_ICB_REPLAY)`?

Reply:
This is used by the regular paths too, I don't think its necessary to set the buffer to null if we're not using it, but actually maybe it is.

Let me see if that triggers a validation error